python/tvm/relax/transform/op_legalizer.py

```diff
@@ -0,0 +1,559 @@
+
+    return bb.call_te(
+        topi.nn.conv2d,
+        input=call.args[0],
+        filter=call.args[1],
+        strides=call.attrs.strides,
+        padding=call.attrs.padding,
+        dilation=call.attrs.dilation,
+        data_layout=call.attrs.data_layout,
+        kernel_layout=call.attrs.kernel_layout,
+        out_dtype=call.attrs.out_dtype if call.attrs.out_dtype != "" else None,
+    )
+
+
+def _add(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.add, call.args[0], call.args[1])
+
+
+def _subtract(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.subtract, call.args[0], call.args[1])
+
+
+def _multiply(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.multiply, call.args[0], call.args[1])
+
+
+def _divide(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.divide, call.args[0], call.args[1])
+
+
+def _floor_divide(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.floor_divide, call.args[0], call.args[1])
+
+
+def _sin(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.sin, call.args[0])
+
+
+def _cos(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.cos, call.args[0])
+
+
+def _tanh(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.tanh, call.args[0])
+
+
+def _negative(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.negative, call.args[0])
+
+
+def _log(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.log, call.args[0])
+
+
+def _sqrt(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.sqrt, call.args[0])
+
+
+def _sigmoid(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.sigmoid, call.args[0])
+
+
+def _less(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.less, call.args[0], call.args[1])
+
+
+def _nn_relu(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.nn.relu, call.args[0])
+
+
+def _nn_gelu(bb: BlockBuilder, call: Call):
+    def gelu(x):
+        dtype = x.dtype
+        return x * (
+            tvm.tir.const(0.5, dtype)
+            + topi.erf(x * tvm.tir.const(0.5**0.5, dtype)) * tvm.tir.const(0.5, dtype)
+        )
+
+    return bb.call_te(gelu, call.args[0], primfunc_name_hint="gelu")
+
+
+def _nn_silu(bb: BlockBuilder, call: Call):
+    sig = bb.emit_te(topi.sigmoid, call.args[0])
+    return bb.call_te(topi.multiply, call.args[0], sig)
+
+
+def _reshape(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.reshape, call.args[0], call.args[1])
+
+
+def _permute_dims(bb: BlockBuilder, call: Call):
+    return bb.call_te(topi.transpose, call.args[0], call.attrs.axes)
+
+
+def _concat(bb: BlockBuilder, call: Call):
+    t = call.args[0]
+    n_field = len(t.struct_info.fields)
+    while isinstance(t, Var):
+        binding = bb.lookup_binding(t)
+        if not isinstance(binding, (Tuple, Var)):
+            break
+        t = binding
+
+    assert isinstance(t, (Tuple, Var))
+    fields = t.fields if isinstance(t, Tuple) else [TupleGetItem(t, i) for i in range(n_field)]
```

Comment:
The result could be reproduced in the following script:
```
from tvm.script.parser import ir as I, relax as R
from tvm.relax.transform import OperatorLegalizer

@I.ir_module
class Mod:
    @R.function
    def main(x: R.Tuple(R.Tensor((3, 4), "float32"), R.Tensor((3, 4), "float32"))):
        gv1 = R.concat(x)
        return gv1


Mean.show()
lowered_mod = OperatorLegalizer(Mod).transform()
lowered_mod.show()
```

Maybe we could use the normalized TupleGetItem node?
```suggestion
    fields = (
        t.fields
        if isinstance(t, Tuple)
        else [bb.normalize(TupleGetItem(t, i)) for i in range(n_field)]
    )
```

Reply:
Another high-level issue: Could we deduce StructInfo for leaf nodes when it is possible?

Reply:
Will emit the TupleGetItems first.